kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java

```diff
@@ -0,0 +1,375 @@
+          return BinaryType.get();
+        case INT8:
+        case INT16:
+          return IntegerType.get();
+        case INT32:
+          if (valueSchema.name() != null) {
+            if (valueSchema.name().equals(Date.LOGICAL_NAME)) {
+              return DateType.get();
+            } else if (valueSchema.name().equals(Time.LOGICAL_NAME)) {
+              return TimeType.get();
+            }
+          }
+          return IntegerType.get();
+        case INT64:
+          if (valueSchema.name() != null && valueSchema.name().equals(Timestamp.LOGICAL_NAME)) {
+            return TimestampType.withZone();
+          }
+          return LongType.get();
+        case FLOAT32:
+          return FloatType.get();
+        case FLOAT64:
+          return DoubleType.get();
+        case ARRAY:
+          Type elementType = toIcebergType(valueSchema.valueSchema());
+          if (config.schemaForceOptional() || valueSchema.valueSchema().isOptional()) {
+            return ListType.ofOptional(nextId(), elementType);
+          } else {
+            return ListType.ofRequired(nextId(), elementType);
+          }
+        case MAP:
+          Type keyType = toIcebergType(valueSchema.keySchema());
+          Type valueType = toIcebergType(valueSchema.valueSchema());
+          if (config.schemaForceOptional() || valueSchema.valueSchema().isOptional()) {
+            return MapType.ofOptional(nextId(), nextId(), keyType, valueType);
+          } else {
+            return MapType.ofRequired(nextId(), nextId(), keyType, valueType);
+          }
+        case STRUCT:
+          List<NestedField> structFields =
+              valueSchema.fields().stream()
+                  .map(
+                      field ->
+                          NestedField.of(
+                              nextId(),
+                              config.schemaForceOptional() || field.schema().isOptional(),
+                              field.name(),
+                              toIcebergType(field.schema())))
+                  .collect(toList());
+          return StructType.of(structFields);
+        case STRING:
+        default:
+          return StringType.get();
+      }
+    }
+
+    Optional<Type> inferIcebergType(Object value) {
+      return Optional.ofNullable(doInferIcebergType(value));
+    }
+
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    private Type doInferIcebergType(Object value) {
+      if (value == null) {
+        return null;
+      } else if (value instanceof String) {
+        return StringType.get();
+      } else if (value instanceof Boolean) {
+        return BooleanType.get();
+      } else if (value instanceof BigDecimal) {
+        BigDecimal bigDecimal = (BigDecimal) value;
+        return DecimalType.of(bigDecimal.precision(), bigDecimal.scale());
+      } else if (value instanceof Number) {
+        Number num = (Number) value;
+        Double dbl = num.doubleValue();
+        if (dbl.equals(Math.floor(dbl))) {
```

Comment:
Won't this coerce `10.0` to an integral type? That doesn't seem right to me.

Reply:
To be more clear, I think I'd check for Integer, Long, Float, Double, etc. separately.

Reply:
Yes, this was unneccessary. Connect internally handles converting to either Long or Double, so we can just use that type directly. I updated that.